daal4py/sklearn/_utils.py

```diff
@@ -75,23 +76,24 @@ def get_daal_version() -> DaalVersionTuple:
     return int(dv()[0:4]), str(dv()[10:11]), int(dv()[4:8])
 
 
+@functools.lru_cache(maxsize=256, typed=False)
 def daal_check_version(
     required_version: Tuple[Any, ...],
-    _get_daal_version: Callable[[], DaalVersionTuple] = get_daal_version,
+    daal_version: Tuple[Any, ...] = get_daal_version(),
```

Comment:
There's a subtle change here. Previously we called `get_daal_version()` on runtime, i.e. when the function `daal_check_version()` is called. With your change this is changed to load time. In particular, `get_daal_version()` is only executed once, and that is when the module is loaded.
Right now, I don't see how this is an issue, as `get_daal_version()` normally produces a constant result. But just wanted to double check that you considered this subtlety.

Reply:
daal version is not expected to change during process life, so `get_daal_version` has static result.

Reply:
Agree, just wanted to mention it :)